[David Zhu] iP

Dethada opened the pull request:
# Cappy
> Hello I'm here to help you manage your tasks! - Cappy

Do you have a **hard time**
- managing your tasks?
- dealing with GUI software?

If you do Cappy is the _perfect_ companion for you!
~Forget your current task management software.~

## Setup instructions 🛠️
1. Download the [jar](https://github.com/Dethada/ip/releases/tag/A-Jar)
2. Make sure java 11 is installed

## Usage 📖
Run it by double clicking on it. Or run `java -jar cappy.jar` in the terminal.

```
 ██████╗ █████╗ ██████╗ ██████╗ ██╗   ██╗
██╔════╝██╔══██╗██╔══██╗██╔══██╗╚██╗ ██╔╝
██║     ███████║██████╔╝██████╔╝ ╚████╔╝
██║     ██╔══██║██╔═══╝ ██╔═══╝   ╚██╔╝
╚██████╗██║  ██║██║     ██║        ██║
 ╚═════╝╚═╝  ╚═╝╚═╝     ╚═╝        ╚═╝                                                                                          
    ____________________________________________________________                                                          
      Hello! I'm Cappy
      What can I do for you?
    ____________________________________________________________
```

## Developer Guide
Entry point of the application.
```java
public static void main(String[] args) {
    UI.showBanner();
    UI.showGreetings();
    try {
        storage = new Storage(STORAGE_PATH);
        TASKS = TaskList.load(storage);
        inputLoop();
        storage.close();
    } catch (IOException | CappyException e) {
        UI.showError(e);
    } finally {
        Cappy.UI.close();
    }
}
```

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Dethada committed:
Updated gitignore

Dethada committed:
Renamed chatbot and finish initial skeletal version

Dethada committed:
Echo user input
- created a print function to handle output formatting

Dethada committed:
Edited print method to take String[] instead

Dethada committed:
Fix chatbot name

Dethada committed:
Able to add and list tasks now

Dethada committed:
Mark tasks as done
- Switch from stroing tasks in ArrayList<String> to Task and TaskList
- Added ability to mark tasks as done

Dethada committed:
Add ability to unmark tasks

Dethada committed:
Add indentations to output

Dethada committed:
Added new task types: Todos, Events and Deadlines

Dethada committed:
Convert bye and list commands to static fields too

Dethada committed:
Added print(String msg) and removed Duke.* prefix

Dethada committed:
use the stored HORIZONTAL_LINE and INDENT

Dethada committed:
Switch to use print(String msg)

Dethada committed:
Fix compilation errors

Dethada committed:
Add "Here are the tasks in your list:" to output of list command

Dethada committed:
Added input and expected output for testing

Dethada committed:
command prefix -> command

Dethada committed:
Handle errors for mark/unmark commands

Dethada committed:
Use custom exception class DukeException

Dethada committed:
Updated test files

Dethada committed:
Reformat

Dethada committed:
Handle errors for todo

Dethada committed:
Updated test files

Dethada committed:
Handle errors for deadline and event

Dethada committed:
Updated test files

Dethada committed:
Added delete command

Dethada committed:
Updated test files

Dethada committed:
Save tasks to disk

Dethada committed:
New feature: Save to and load task list from disk

Dethada committed:
Updated test script to remove generated data

Dethada committed:
Merge branch 'branch-Level-7'

Dethada committed:
Parse datetime for datetime inputs

Datetime inputs was previously stored as a String.

Dethada committed:
Merge branch 'branch-Level-8'

Dethada committed:
Split into more classes

Dethada committed:
Merge branch 'A-MoreOOP'

Dethada committed:
Organize into Packages

Dethada committed:
Update test script to handle new dir structure

Dethada committed:
Merge branch 'A-Packages'

Dethada committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Dethada committed:
Update class path

Dethada committed:
Add a dummy junit test class

Dethada committed:
Add junit tests

- Also update gitignore to ignore gradle generated files

Dethada committed:
Merge branch 'junit'

Dethada committed:
Update Jar filename

Dethada committed:
Auto format

Dethada committed:
Avoid unnecessary use of this with fields

Dethada committed:
Add javadocs for main source code

Dethada committed:
Add find command

Dethada committed:
Merge branch 'branch-A-Javadoc'

Dethada committed:
Merge branch 'branch-A-CodingStandard'

Dethada committed:
Merge branch 'branch-Level-9'

Dethada committed:
Rename boolean variables

Dethada committed:
Allow multi word descriptions

Dethada committed:
Change HashMap to Map

Dethada committed:
Add javadoc for TaskList::search

Dethada committed:
Add checkstyle to gradle

Dethada committed:
Update checkstyle config

Dethada committed:
Reformat to pass checkstyle

Dethada committed:
Add spotless to autoformat code

Dethada committed:
Merge branch 'A-CheckStyle'

Dethada committed:
Remove text-ui-test

Not needed for GUI applications

Dethada committed:
Initial javafx setup from tutorial

cliftonchee commented on src/main/java/cappy/command/EventCommand.java:
```diff
@@ -0,0 +1,51 @@
+        } else if (!input.hasNamedArgument("from") || !input.hasNamedArgument("to")) {
+            throw new CappyException(
+                    "Please specify the duration of the event using /from [Date Time] /to [Date"
+                            + " Time].");
```
Good job on indentation for lines that are too long!

e.g.

```java
if (isReady) {
    setText("Long line split"
            + "into two parts.");
}
```

cliftonchee commented on src/main/java/cappy/storage/CsvFormat.java:
```diff
@@ -0,0 +1,7 @@
+public interface CsvFormat {
+    /** Returns the CSV string representation of this object. */
+    public String toCsv();
+}
```
Good job on not falling into the pit trap of naming it CSV instead!

cliftonchee commented on src/test/java/cappy/parser/ParsedInputTest.java:
```diff
@@ -0,0 +1,78 @@
+
+public class ParsedInputTest {
+    @Test
+    public void numberOfPositionalArguments() {
```
Understand that it is already labelled with "@Test", but maybe consider adding Test to the function name

e.g.

```java
  @Test
  public void anotherDummyTest(){
      assertEquals(4, 4);
  }
```

cliftonchee commented on src/main/java/cappy/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    /** Sets the status of the task to be done (complete). */
+    public void done() {
```
I understand that it is clear that this function sets the task to be done, but can be further clarified for function name for setters.

e.g. 
`void setFound(boolean isFound)`

cliftonchee commented on src/main/java/cappy/task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+                + ","
+                + Parser.dateTimeToString(due)
+                + ",";
+    }
```
I like that it is neatly organised. Good job!

cliftonchee reviewed (commented):
Looks good to me! Neatly organised and well-explained structure. Easy to read!

Dethada committed:
Modify GUI to have TUI capabilities

Dethada committed:
Change profile pictures

Dethada committed:
Merge branch 'branch-Level-10'

Dethada commented:
Thanks @cliftonchee for the review 😄

Dethada committed:
Add cappy.csv to gitignore

Dethada committed:
Added assertions

- cappy.controller.MainWindow.setCappy: cappy cannot be set to null
- cappy.TaskList.getTask: index provided must be valid
- cappy.TaskList.removeTask: index provided must be valid
- cappy.TaskList.load: storage must not be null

Dethada committed:
Improve code quality

- reduce the use of Magic literals
- add default branch for more conditionals

Dethada committed:
Merge pull request #1 from Dethada/branch-A-Assertions

Added assertions

Dethada committed:
Merge branch 'master' into branch-A-CodeQuality

Dethada committed:
Merge pull request #2 from Dethada/branch-A-CodeQuality

Improve code quality

Dethada committed:
Use GitHub Actions to set up CI

Dethada committed:
Add a new command `help`

Dethada committed:
Fix resizing

Dethada committed:
Fix text truncation

Dethada committed:
Add credits for storage constructor code

Dethada committed:
Set title bar text to Cappy

Dethada committed:
Add product screenshot

Dethada committed:
Update user guide

Dethada committed:
Handle exit correctly when done via GUI
